backend/internal/storage/store.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type Storer interface {
+	ProcessDeployed(
+		processDefinitionKey int64,
+		bpmnProcessID string,
+		version int64,
+		deploymentTime time.Time,
+		bpmnResourceRaw []byte,
+	) error
+
+	ProcessInstanceActivated(
+		processInstanceKey int64,
+		processDefinitionKey int64,
+		version int64,
+		startTime time.Time,
+	) error
+
+	ProcessInstanceCompleted(
+		processInstanceKey int64,
+		endTime time.Time,
+	) error
+
+	ProcessInstanceTerminated(
+		processInstanceKey int64,
+		endTime time.Time,
+	) error
+
+	VariableCreated(
+		processInstanceKey int64,
+		name string,
+		value string,
+		time time.Time,
+	) error
+
+	VariableUpdated(
+		processInstanceKey int64,
+		name string,
+		value string,
+		time time.Time,
+	) error
+}
+
+// TODO: use context for queries where reasonable
+
+type databaseStorer struct {
+	db *gorm.DB
+}
+
+func NewStorer(db *gorm.DB) Storer {
+	return &databaseStorer{db}
+}
+
+// call this for each processesMetadata
+func (r *databaseStorer) ProcessDeployed(
+	processDefinitionKey int64,
+	bpmnProcessID string,
+	version int64,
+	deploymentTime time.Time,
+	bpmnResourceRaw []byte,
+) error {
+	bpmnResource := BpmnResource{
+		BpmnProcessID: bpmnProcessID,
+		BpmnFile:      base64.StdEncoding.EncodeToString(bpmnResourceRaw),
+	}
+
+	err := r.db.Create(&Process{
+		ProcessDefinitionKey: processDefinitionKey,
+		BpmnProcessID:        bpmnProcessID,
+		Version:              version,
+		DeploymentTime:       deploymentTime,
+		BpmnResource:         bpmnResource,
+	}).Error
+	if err != nil {
+		return fmt.Errorf("failed to create process: %w", err)
+	}
+
+	return nil
+}
+
+func (r *databaseStorer) ProcessInstanceActivated(
+	processInstanceKey int64,
+	processDefinitionKey int64,
+	version int64,
+	startTime time.Time,
+) error {
+	err := r.db.Create(&Instance{
+		ProcessInstanceKey:   processInstanceKey,
+		ProcessDefinitionKey: processDefinitionKey,
+		Version:              version,
+		Status:               "ACTIVE",
```

Comment:
This status enum is defined in generated code so I don't really see a solution to this. I guess we could just revert the GraphQL enums back to strings. I would've just liked it to be an enum since it clearly is internally an enum, but guess we can't have nice things.